### DISCUSSION TITLE
A little guidance

### DISCUSSION THREAD

**Discussion by Taha-Firoz** [samuelmaddock/electron-browser-shell#21] [General]
First of all, amazing job on this project! The code is well written and super easy to read and understand. I've been thinking about doing something like this for a while but really didn't understand how to go about it before this.

Could you help me out a little bit by answering a couple of questions I have about something that I'm trying to implement?

Basically I'm only interested in rendering a react app that I've made as the only window that's always open but I want external link to open in a separate tab with an option to close the tab or go back to the main react app. Also I'd like to view the list of opened tabs right in the react app, like basically making it a home screen. 

So in order to implement that I think I'd need to:

1. Pull out features like extensions since I don't really need them, how would I go about making this UI shell even leaner so it's easier to reason around.

2. Can I replace the webui with The react app I have in mind and maybe add in some Tailwind instead of raw CSS?

The main reason I asked 2 is I see you're injecting the chrome object into the webui module. I'm just confused as how one would go about that in am exported react app?

**Comment by samuelmaddock** [comment]
The shell package in this project was built solely for the purpose of testing extensions. For that reason, I chose to avoid introducing any frameworks as the UI wasn't what's important to me.

That said, it's certainly possible to repurpose the project however you'd like, but I won't be able to tell you the best way to architect it.

I'm injecting the `chrome.*` extension APIs into the webui as a convenient hack for managing the state of browser tabs. Getting the list of tabs and interacting with them uses the `chrome.tabs` API. By foregoing the extensions support, you'd need to design your own IPC methods for interacting with tabs in Electron's main process.
